src/formats/apple-notes.ts

```diff
@@ -251,8 +284,42 @@ export class AppleNotesImporter extends FormatImporter {
 
 		const folder = this.resolvedFolders[row.ZFOLDER] || this.rootFolder;
 
-		const title = `${row.ZTITLE1}.md`;
-		const file = await this.saveAsMarkdownFile(folder, title, '');
+		// Get creation date and format it according to user preference
+		let title = row.ZTITLE1;
+		if (this.filePrefixFormat) {
+			const creationDate = new Date(this.decodeTime(row.ZCREATIONDATE3 || row.ZCREATIONDATE2 || row.ZCREATIONDATE1));
+			const datePrefix = this.filePrefixFormat
+				.replace('YYYY', creationDate.getUTCFullYear().toString())
+				.replace('MM', (creationDate.getUTCMonth() + 1).toString().padStart(2, '0'))
+				.replace('DD', creationDate.getUTCDate().toString().padStart(2, '0'));
+			title = `${datePrefix} ${title}`;
+		}
+
+		const fullPath = `${folder.path}/${title}.md`;
+
+		// Check for duplicate notes based on the selected handling option
+		const existingFile = this.vault.getAbstractFileByPath(fullPath) as TFile;
+		
+		if (existingFile) {
```

Comment:
Instead of asserting above, please do `if (existingFile && existingFile instanceof TFile) {`